src/database_lmdb.cpp

```diff
@@ -0,0 +1,341 @@
+/*
+Minetest
+Copyright (C) 2013 celeron55, Perttu Ahola <celeron55@gmail.com>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#include <vector>
+#include <string>
+#include "irr_v3d.h"
+#include "irrlichttypes.h"
+#include "database.h"
+#include "database_lmdb.h"
+#include "settings.h"
+#include "exceptions.h"
+#include "lmdb.h"
+#include "filesys.h"
+#include <errno.h>
+
+#define MAKEKEY(pos)                    \
+	MDB_val key;                        \
+	s64 key_ = getBlockAsInteger(pos);  \
+	key.mv_size = sizeof(s64);          \
+	key.mv_data = &key_;
+
+//public:
+Database_LMDB::Database_LMDB(const std::string &savedir, Settings &conf)
+{
+	int result;
+
+	online = false;
+	path = std::string (savedir) + DIR_DELIM + "map.mdb";
+	configuration = &conf;
+
+	mapSize = conf.exists("lmdb_mapsize") ? conf.getU64("lmdb_mapsize") :
```

Comment:
If a value can be specified in MB instead of bytes, then in practise, it is always a multiple of any OS's page size (at least, for the known, common CPU architectures). The call to `mdb_env_set_mapsize` would multiply the value by 1MB:
```
mdb_env_set_mapsize(environment, mapSizeInMB * 1024 * 1024)
```